packages/nocodb/src/plugins/gcs/Gcs.ts

```diff
@@ -176,13 +177,27 @@ export default class Gcs implements IStorageAdapterV2 {
 
     const [url] = await this.storageClient
       .bucket(this.bucketName)
-      .file(key)
+      .file(this.patchKey(key))
       .getSignedUrl(options);
 
     return url;
   }
 
-  public async scanFiles(_globPattern: string): Promise<Readable> {
-    return Promise.resolve(undefined);
+  public async scanFiles(globPattern: string): Promise<Readable> {
+    const stream = new Readable({
+      objectMode: true,
+      read() {},
+    });
+
+    const [files] = await this.storageClient.bucket(this.bucketName).getFiles({
+      prefix: globPattern.replace(/^\*/, ''),
+    });
```

Comment:
Would this work for huge number of files? Even so it is not a good idea to load all into memory at once.
Can we use getFilesStream instead?